src/overlays/actors/ovl_En_Crow/z_en_crow.c

```diff
@@ -88,60 +103,463 @@ static DamageTable D_8099C0AC = {
+        this->actor.home.pos.x = this->actor.parent->world.pos.x;
+        this->actor.home.pos.z = this->actor.parent->world.pos.z;
+        sp38 = Actor_XZDistanceToPoint(&this->actor, &this->actor.parent->world.pos);
+    } else {
+        sp38 = 450.0f;
+        this->actor.flags |= ACTOR_FLAG_1;
+    }
+
+    if (this->actor.bgCheckFlags & 8) {
+        this->aimRotY = this->actor.wallYaw;
+    } else if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 300.0f) {
+        this->aimRotY = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    }
+
+    if ((Math_SmoothStepToS(&this->actor.shape.rot.y, this->aimRotY, 5, 0x300, 0x10) == 0) && (skelanimeUpdated) &&
+        (Rand_ZeroOne() < 0.1f)) {
+
+        temp = (Actor_YawToPoint(&this->actor, &this->actor.home.pos) - this->actor.shape.rot.y);
+        if (temp > 0) {
+            this->aimRotY += Rand_S16Offset(0x1000, 0x1000);
+        } else {
+            this->aimRotY -= Rand_S16Offset(0x1000, 0x1000);
+        }
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_KAICHO_CRY);
+    }
+
+    if ((this->actor.depthInWater > -40.0f) || (this->actor.bgCheckFlags & 1)) {
+        this->aimRotX = -0x1000;
+    } else if (this->actor.world.pos.y < (this->actor.home.pos.y - 50.0f)) {
+        this->aimRotX = -Rand_S16Offset(0x800, 0x800);
+    } else if (this->actor.world.pos.y > (this->actor.home.pos.y + 50.0f)) {
+        this->aimRotX = Rand_S16Offset(0x800, 0x800);
+    }
+
+    if ((Math_SmoothStepToS(&this->actor.shape.rot.x, this->aimRotX, 0xA, 0x100, 8) == 0) && skelanimeUpdated &&
+        (Rand_ZeroOne() < 0.1f)) {
+        if (this->actor.home.pos.y < this->actor.world.pos.y) {
+            this->aimRotX -= Rand_S16Offset(0x400, 0x400);
+        } else {
+            this->aimRotX += Rand_S16Offset(0x400, 0x400);
+        }
+
+        this->aimRotX = CLAMP(this->aimRotX, -0x1000, 0x1000);
+    }
+
+    if ((this->actor.bgCheckFlags & 1) != 0) {
+        Math_ScaledStepToS(&this->actor.shape.rot.x, -0x100, 0x400);
+    }
+
+    DECR(this->timer);
+    if ((this->timer == 0) &&
+        (((this->actor.xzDistToPlayer < 300.0f) && !(player->stateFlags1 & 0x800000)) || (sp38 < 300.0f)) &&
+        (this->actor.depthInWater < -40.0f) && (Player_GetMask(globalCtx) != PLAYER_MASK_STONE)) {
+        if (sp38 < this->actor.xzDistToPlayer) {
+            this->actor.child = this->actor.parent;
+        } else {
+            this->actor.child = &player->actor;
+        }
+        EnCrow_SetupDiveAttack(this);
+    }
+}
+
+void EnCrow_SetupDiveAttack(EnCrow* this) {
+    this->timer = 300;
+    this->actionFunc = EnCrow_DiveAttack;
+    this->actor.speedXZ = 4.0f;
+    this->skelAnime.playSpeed = 2.0f;
+}
+
+void EnCrow_DiveAttack(EnCrow* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 facingPlayer;
+    Vec3f pos;
+    s16 target;
+    s16 phi_a1;
+
+    SkelAnime_Update(&this->skelAnime);
+    DECR(this->timer);
```

Comment:
Not sure the reasoning, but iirc generally we don't use `DECR` outside of conditionals, so this should probably be expanded out.